generative/3d_ldm/utils.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from datetime import timedelta
+
+import numpy as np
+import torch
+import torch.distributed as dist
+from monai.apps import DecathlonDataset
+from monai.bundle import ConfigParser
+from monai.data import DataLoader
+from monai.transforms import (
+    AddChanneld,
+    CenterSpatialCropd,
+    Compose,
+    EnsureChannelFirstd,
+    EnsureTyped,
+    Lambdad,
+    LoadImaged,
+    Orientationd,
+    RandSpatialCropd,
+    ScaleIntensityRangePercentilesd,
+    Spacingd,
+)
+
+
+def setup_ddp(rank, world_size):
+    print(f"Running DDP diffusion example on rank {rank}/world_size {world_size}.")
+    print(f"Initing to IP {os.environ['MASTER_ADDR']}")
+    dist.init_process_group(
+        backend="nccl", init_method="env://", timeout=timedelta(seconds=36000), rank=rank, world_size=world_size
+    )  # gloo, nccl
+    dist.barrier()
+    device = torch.device(f"cuda:{rank}")
+    return dist, device
+
+
+def prepare_dataloader(args, batch_size, patch_size, randcrop=True, rank=0, world_size=1, cache=1.0, download=False):
+    # %% [markdown]
```

Comment:
May consider cleaning the docstring here?

Reply:
Thanks!